website/server/controllers/top-level/payments/iap.js

```diff
@@ -92,88 +87,51 @@ api.iapiOSVerify = {
   url: '/iap/ios/verify',
   middlewares: [authWithHeaders()],
   async handler (req, res) {
+
+    let appleRes = await applePayments.verifyGemPurchase(res.locals.user, req.body.transaction.receipt, req.headers);
```

Comment:
if req.body.transaction doesn't exist it'll crash, should probably have a validation of the body